pkg/ebpf/perf.go

```diff
@@ -68,14 +60,24 @@ func (c *PerfHandler) RecordHandler(record *perf.Record, _ *manager.PerfMap, _ *
 		return
 	}
 
-	c.DataChannel <- &DataEvent{CPU: record.CPU, Data: record.RawSample, r: record}
+	c.dataChannel <- &DataEvent{CPU: record.CPU, Data: record.RawSample, pr: record}
+}
+
+// DataChannel returns the channel with event data
+func (c *PerfHandler) DataChannel() <-chan *DataEvent {
+	return c.dataChannel
+}
+
+// LostChannel returns the channel with lost events
+func (c *PerfHandler) LostChannel() <-chan uint64 {
+	return c.lostChannel
 }
 
 // Stop stops the perf handler and closes both channels
 func (c *PerfHandler) Stop() {
 	c.once.Do(func() {
 		c.closed = true
```

Comment:
I think that's a little bit out of scope - but that's racy (along  with RecordHandler)

Reply:
... all that to say that I'd favor removing the `c.closed` variable altogether and add a comment that it's the callers responsibility to terminate the producers before calling `Stop()`